### PR TITLE
Make IKOS available in the spaceros image (#99).

### PR DESCRIPTION
**Pull request by ivanperez-keera**
IKOS is one of the tools that Space ROS includes to perform static analysis of ROS 2 applications. Although ament_ikos is provided with the docker image, IKOS itself is not, rendering the former useless and leaving it to users to compile and install IKOS prior to being able to use it.

This commit modifies the spaceros Earthfile to install IKOS globally in the system.

### CODE REVIEW

**Comment by ivanperez-keera** [comment]
Filing this as a draft. I built it locally and checked that ikos was building and running during compilation. I'd like to check that it's available in the result. I could use a couple of extra eyes. It does not take a hugely long time to compile.

File: spaceros/Earthfile
Comment (xfiderek) [inline spaceros/Earthfile:155]: Imo we should clone it the same way as it is done for cobra, by creating `ikos_vendor` package in `ament_ikos` repo. See: https://github.com/ament/ament_cobra/tree/master/cobra_vendor.

Thoughts?

Comment (ivanperez-keera) [reply spaceros/Earthfile:155]: I'm worried that it might take quite a bit to prepare that. Note that the repo is removed right after installation.

I'll also note that we are also looking into using DEBs in the future, so I'm not too worried because this installation procedure is temporary, just to at least have something. (I could provide DEBs now but I don't know where they should be housed, especially since we expect Debian and hopefully also Ubuntu to make IKOS available in their repos.)

**Comment by ivanperez-keera** [comment]
This is ready for review, if anyone wants to take it.

**Comment by xfiderek** [comment]
Have you checked if https://github.com/space-ros/docker/issues/104 is fixed by this PR? If not i can dig into artifact output and check.

**Comment by ivanperez-keera** [comment]
@xfiderek I have not, and I don't want to add that to this release. We are getting dangerously close; I think it's best that we first close the items assigned to this release that we have not finished yet.

**Comment by ivanperez-keera** [comment]
@xfiderek please let me know if all is ok. If you can review this and approve it, that'd be great. It can't be merged until someone approves. Thanks!

**Review by xfiderek — APPROVED** [review]

**Comment by ivanperez-keera** [comment]
Upps. "At least 1 approving review is required by reviewers _with write access_"

**Comment by ivanperez-keera** [comment]
Sorry, @xfiderek . It appears that it needs a review also from someone with write access.

I've asked for an additional review.

Thanks for approving it though :)

**Comment by xfiderek** [comment]
Ok, btw after this PR is merged lets create an issue to track status of apt availability, something like "install ikos with from deb". Does it make sense?

**Comment by ivanperez-keera** [comment]
Maybe we can have a quick talk about how to best do this. I'm available to talk / call.

We definitely need to track it, but what's odd about this kind of issue is that instead of us doing something, it's in "waiting" or "on hold" until someone else does something. Perhaps it should be a discussion instead of an issue, or it should be a task in my personal agenda as opposed to an issue in the docker.

There's also the possibility that they might say "no" to both accepting IKOS in debian as `non-free`, and changing the license to accept it in `main`. In that case, the issue might not be "install ikos from deb", but something else.

**Comment by ivanperez-keera** [comment]
And just for full context: https://ftp-master.debian.org/new/ikos_3.2-1.html.

It can take months for that to be processed because of the strange license. The debian dev that put that there has been supernice about helping us; we just have to be very patient.

**Review by EzraBrooks — APPROVED** [review]

**Comment by xfiderek** [comment]
@ivanperez-keera sorry, I could not reply earlier. Personally I am fine with whatever you decide. Since you are actively involved in IKOS dev, we have first-hand news and I find it hard to believe that it will slip out of your mind :).

**Pull request merged**
